Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2610,6 +2680,113 @@ namespace Babylon
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeStarts,
+                std::set<xr::System::Session::Frame::InputSource::Identifier> squeezeEnds,
+                Napi::Env env)
+            {
+                for (const auto& id : selectStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SELECT_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSelects.insert(id);
+                    }
+                }
+
+                for (const auto& id : squeezeStarts)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SQUEEZE_START)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSqueezes.insert(id);
+                    }
+                }
+
+                for (const auto& id : selectEnds)
+                {
+                    auto inputSourceFound = m_idToInputSource.find(id);
+                    if (inputSourceFound != m_idToInputSource.end())
+                    {
+                        auto inputSourceVal = inputSourceFound->second.Value();
+                        Napi::Object inputSourceEvent = GenerateXRInputSourceEvent(inputSourceVal, env);
+
+                        for (const auto& [name, callback] : m_eventNamesAndCallbacks)
+                        {
+                            if (name == JS_EVENT_NAME_SELECT_END)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+
+                            if (name == JS_EVENT_NAME_SELECT)
+                            {
+                                callback.Call({inputSourceEvent});
+                            }
+                        }
+
+                        m_activeSelects.erase(id);
```

Comment:
>id [](start = 46, length = 2)

nit: rather than using `id` here, use `inputSourceFound`. If you use the `std::map<>::key_type` overload it's likely going to have to internally execute a `find` to locate the node to erase. Passing in the iterator position should be slightly faster.

Reply:
That's an iterator for a different object, unless I read this wrong